### PR TITLE
Add solution 1 with the unittests

### PR DESCRIPTION
**Pull request by MohammadRAlSalloum**
---
name: solution review 
about: A template PR for code review with a checklist
---

## Behavior

A guessing game that compares the user's input to randomly generated int.
The user sets boundaries of the generated int.

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/guessing_game.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        
+
+"""
+lower_bound =input("Enter the lower bound: ")
+lower_bound =int(lower_bound)
+upper_bound =input("Enter the upper bound: ")
+upper_bound = int(upper_bound)
+random_number =random.randint(lower_bound, upper_bound)
+guess =input(f"Guess a number between {lower_bound} and {upper_bound}: ")
+guess =int(guess)
```
Comment (RamaBio20) [inline solutions/guessing_game.py:25]: py formatting error

Comment (MohammadRAlSalloum) [reply solutions/guessing_game.py]: Solved

File: solutions/guessing_game.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+"""Guessing Game
+write a function that takes a random integer from the
+user (the player) and compare it to a randomly generated number inside
+the function that is bound by user lower and upper boundaries.
+"""
```
Comment (RaedAlyan) [inline solutions/guessing_game.py:5]: @MohammadRAlSalloum please add the author and the created date here

File: solutions/guessing_game.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    Raises:
+        AssertionError: if the lower_bound argument is not an int.
+        AssertionError: if the upper_bound argument is not an int.
+
+    >>> guessing_game(2, 5)
+        You Won! (if the user's guess matched the random int)
+    >>> guessing_game(-3, 6)
+        Try Again (if the user's guess did
+                    not match the random int)
+
```
Comment (RaedAlyan) [inline solutions/guessing_game.py:32]: @MohammadRAlSalloum please add a new doctest here

File: solutions/guessing_game.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    """
+
+
+lower_bound = input("Enter the lower bound: ")
+lower_bound = int(lower_bound)
```
Comment (RaedAlyan) [inline solutions/guessing_game.py:38]: @MohammadRAlSalloum within the docstring of your function, we mentioned that there is an AssertionError if the input isn't integer, please add your checker for this purpose

File: solutions/guessing_game.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
+"""Guessing Game
+write a function that takes a random integer from the
+user (the player) and compare it to a randomly generated number inside
+the function that is bound by user lower and upper boundaries.
+"""
+
+import random
+
+
+def guessing_game(lower_bound: int, upper_bound: int) -> int:
```
Comment (RaedAlyan) [inline solutions/guessing_game.py:10]: @MohammadRAlSalloum you defined this function and didn't use it, please correct me if I'm wrong :)

File: solutions/guessing_game.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        Try Again (if the user's guess did
+                    not match the random int)
+
+
+    """
+
+
+lower_bound = input("Enter the lower bound: ")
+lower_bound = int(lower_bound)
+upper_bound = input("Enter the upper bound: ")
```
Comment (RaedAlyan) [inline solutions/guessing_game.py:39]: @MohammadRAlSalloum In my opinion, I prefer to don't let the user input values because when you implement your test for this challenge, the values will be passed already to the function.

File: solutions/guessing_game.py (hunk 2 above)
Comment (MohammadRAlSalloum) [reply solutions/guessing_game.py:8]: Done. Thank you

File: solutions/guessing_game.py (hunk 4 above)
Comment (MohammadRAlSalloum) [reply solutions/guessing_game.py]: Done thank you

File: solutions/guessing_game.py (hunk 6 above)
Comment (MohammadRAlSalloum) [reply solutions/guessing_game.py]: Done Thank you

File: solutions/tests/test_guessing_game.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+                        when the user's guess does not match the random number.
+    test_boundary_condition: Tests if the function works correctly for the boundary condition
+                            where the lower and upper bounds are the same.
+
+Usage:
+    Run this module with a Python test runner to execute the unittests.
+"""
+
+import unittest
+from unittest.mock import patch
```
Comment (RaedAlyan) [inline solutions/tests/test_guessing_game.py:26]: good job @MohammadRAlSalloum,
you can write your test methods without using patch

**Review by RamaBio20 — COMMENTED** [review]
this PR looks good

File: solutions/tests/__init__.py
Comment (RaedAlyan) [inline solutions/tests/__init__.py:1]: please remove this file because I've pushed it before

File: solutions/__init__.py
[the remark this answers is not in the document]
Comment (MohammadRAlSalloum) [reply solutions/__init__.py:1]: removing it will not allow the ci checks to be completed

File: solutions/tests/__init__.py
Comment (MohammadRAlSalloum) [reply solutions/tests/__init__.py:1]: removing it will not allow the ci checks to be completed
